crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1814,6 +1790,246 @@ enum Dependencies {
     Available(Vec<(PubGrubPackage, Range<Version>)>),
 }
 
+impl Dependencies {
+    fn fork(self) -> ForkedDependencies {
+        use std::collections::hash_map::Entry;
+
+        let deps = match self {
+            Dependencies::Available(deps) => deps,
+            Dependencies::Unavailable(err) => return ForkedDependencies::Unavailable(err),
+        };
+
+        let mut by_name: FxHashMap<&PackageName, PossibleForks> = FxHashMap::default();
+        for (index, (ref pkg, _)) in deps.iter().enumerate() {
+            let (name, marker) = match &**pkg {
+                // A root can never be a dependency of another package, and a `Python` pubgrub
+                // package is never returned by `get_dependencies`. So these cases never occur.
+                PubGrubPackageInner::Root(_) | PubGrubPackageInner::Python(_) => unreachable!(),
+                PubGrubPackageInner::Package { name, marker, .. }
+                | PubGrubPackageInner::Extra { name, marker, .. }
+                | PubGrubPackageInner::Dev { name, marker, .. } => (name, marker.as_ref()),
+                PubGrubPackageInner::Marker { name, marker, .. } => (name, Some(marker)),
```

Comment:
I removed the `ref name` and `ref marker` here and now use `marker.as_ref()`, because the `PubGrubPackageInner::Marker` arm needs needs to return `Option<&Marker>` not `&Option<Marker>`.